react-es6-popup/src/popup.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Nested from './nested-component';
+
+class Popup extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>React Component</h1>
+        <p>
+          This is an example of a popup UI in React.
+        </p>
+        <Nested />
+      </div>
+    );
+  }
+}
+
+ReactDOM.render(<Popup/>, document.getElementById('app'));
```

Comment:
it would be nice to make a single WebExtensions API call here and render the component with some data in it (without add too much complexity to the example, e.g. just a single call to `browser.tabs.query` or `browser.storage.local.get/set`)